### PR TITLE
gradle configuration update

### PR DESCRIPTION
**Pull request by UpcraftLP**
- moved version variables to gradle.properties file
- use constants for the compatibility levels
- add a default configuration for the maven-publish plugin
- fix character encoding in some edge cases where UTF-8 is not the system default charset
- add javadoc task

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,23 +1,32 @@
 plugins {
 	id 'fabric-loom' version '0.2.0-SNAPSHOT'
+    id 'maven-publish'
```
Comment (modmuss50) [inline build.gradle]: Bad indentation

File: build.gradle [hunk 2]
```diff
@@ -27,3 +36,49 @@ task sourcesJar(type: Jar, dependsOn: classes) {
 	classifier = 'sources'
 	from sourceSets.main.allSource
 }
+
+// create a javadoc file for publishing
+task javadocJar(type: Jar, dependsOn: javadoc) {
```
Comment (modmuss50) [inline build.gradle]: Do we really need a javadoc?

Comment (UpcraftLP) [reply build.gradle]: well I thought it'd be a nice-to-have, mainly for including it in the maven publication

Comment (modmuss50) [reply build.gradle]: It adds quite a bit more to the build.gradle (due to the fix bellow). And there are source jars anyway so when you depend on a mod it should give you sources.

**Comment by modmuss50** [comment]
Another thing I did want to include in the example mod was auto setting the version number in the fabric.mod.json file.

This can be done like so:

```groovy
processResources {
  from(sourceSets.main.resources.srcDirs) {
    include 'fabric.mod.json'
    expand version: project.version
  }
  from(sourceSets.main.resources.srcDirs) {
    exclude 'fabric.mod.json'
  }
}
```
```json
{
  "id": "modid",
  "name": "Example mod",
  "version": "$version"
}
```

Not a critical thing to get in, would be nice though.

**Comment by UpcraftLP** [comment]
>Another thing I did want to include in the example mod was auto setting the version number in the fabric.mod.json file.

Pretty much what I do in all of my mods anyways, sure, will add it

**Pull request merged**